src/utils/PolylineStitcher.h

```diff
@@ -147,8 +160,15 @@ class PolylineStitcher
                     assert( ! processed[closest.poly_idx]);
                     processed[closest.poly_idx] = true;
                 }
+
                 if (closest_is_closing_polygon)
                 {
+                    if (go_in_reverse_direction)
+                    { // re-reverse chain to retain original direction
+                        // NOTE: not sure if this code could ever be reached, since if a polygon can be closed that should be already possible in the forward direction
+                        chain.reverse();
+                    }
```

Comment:
I missed this from last PR, but allow me to comment here what I should have commented on the polylines PR.

```cpp
            for (bool go_in_reverse_direction : { false, true })
            {
                if (go_in_reverse_direction)
                { // try extending chain in the other direction
                    chain.reverse();
                }
                
                ...
                
```
I don't think this reversing behaviour is very clear as this is dependent on the ordering of the reversal. If you would loop though the ordering like `go_in_reverse_direction : { true, false }` then this trick wouldn't work; the `true` case mutates the chain, and therefor must be done last. 

Why not use `emplace_back` or `emplace_front` conditionally on `go_in_reverse_direction`?

Reply:
If using `emplace_back` / `emplace_front` add too much performance/complexity/overhead then at the very least add a comment that and why the true case must be handled last in the for loop.

Reply:
> Why not use emplace_back or emplace_front conditionally on go_in_reverse_direction?

Because `emplace_front()` has a linear time complexity, rather than the constant time complexity of `emplace_back`.

Ok I've added comments.